config.yaml

```diff
@@ -37,6 +37,39 @@ page_list:
 # Footer Config
 footer_config:
   text: Got questions or want to get involved? We'd love to hear from you! Feel free to contact us at the socials listed or by our  [contact us](Contact) page. We'd love to hear from you and help you get the most out of your experience with the Trent Computer Science Club Association.
+# Configuration for events
+events: &events # Future Events
+  - title: Future Event
+    href: https://www.example.com/
+    open_date: 2024-08-20
+    date: 2024-08-25
+    image:
+      src: https://via.placeholder.com/150
+      alt: placeholder image
+  # Current Events
+  - title: Trent Computer Science Hackathon
+    main_event: true # Note we can only have one of these
```

Comment:
> Since this is the case, lets maybe add some validation to enforce it. Ex.
> 
> ```ts
>   events: z.array(eventItem).refine(
>     (events) => {
>       const mainEvents = events.filter(event => event.main_event);
>       return mainEvents.length <= 1;
>     },
>     {
>       message: 'Only one event can be designated as the main event.',
>       path: ['events'],
>     }
> ```

I didn't even realize we could do this with `zod` I'll get on that (for reference I do currently handle this with a runtime error when building the component).